process/transaction/baseProcess.go

```diff
@@ -223,23 +224,33 @@ func (txProc *baseTxProcessor) VerifyTransaction(tx *transaction.Transaction) er
 
 // Setting a guardian is allowed with regular transactions on a guarded account
 // but in this case is set with the default epochs delay
-func checkOperationAllowedToBypassGuardian(txData []byte) error {
-	if process.IsSetGuardianCall(txData) {
-		return nil
+func (txProc *baseTxProcessor) checkOperationAllowedToBypassGuardian(tx data.TransactionHandler) error {
+	if !process.IsSetGuardianCall(tx.GetData()) {
+		return fmt.Errorf("%w, not allowed to bypass guardian", process.ErrTransactionNotExecutable)
 	}
 
-	return fmt.Errorf("%w, not allowed to bypass guardian", process.ErrTransactionNotExecutable)
+	return txProc.CheckSetGuardianExecutable(tx)
+}
+
+// CheckSetGuardianExecutable checks if the setGuardian builtin function is executable
+func (txProc *baseTxProcessor) CheckSetGuardianExecutable(tx data.TransactionHandler) error {
+	err := txProc.scProcessor.CheckBuiltinFunctionIsExecutable(core.BuiltInFunctionSetGuardian, tx)
```

Comment:
What will happen when such transaction is selected by the leader for scheduled transactions? Is it added in block, occupying blockspace and treated as invalid tx? I think these kinds of transactions should not be selected, nor included in block if we do not consume gas for them.

Reply:
I think an idea is that the proposer should pass through this filter each and every tx it should include in proposed scheduled miniblock. I won't filter out transactions at interceptor time because we can have stuck nodes.

Reply:
should not be selected due to the returned error.
in normal processing the leader does processAndRemoveBadTransaction which handles the ErrTransactionNotExecutable just like ErrLowerNonceInTransaction after CheckSetGuardianExecutable on the ProcessTransaction (through VerifyTransaction).

VerifyTransaction is also called in func (txs *transactions) verifyTransaction which is called from createScheduledMiniBlocks
so these transactions should not reach the scheduled miniblock.